src/instructlab/model/train.py

```diff
@@ -87,29 +159,72 @@
         "(reduces GPU VRAM usage and may slow down training)"
     ),
 )
+@click.option("--max-seq-len", type=int)
+@click.option("--max-batch-len", type=int)
 @click.option(
-    "--model-name",
-    default="instructlab/merlinite-7b-lab",
-    show_default=True,
-    help="model name to use in training",
+    "--effective-batch-size", type=int, help="total batch size across all GPUs"
+)
+@click.option(
+    "--save-samples",
+    type=int,
+    help="how many amples to process before saving a checkpoint",
+)
+@click.option("--learning-rate", type=float, help="learning rate for training")
+@click.option("--warmup-steps", type=int, help="warmup steps for training")
+@click.option(
+    "--deepspeed-config",
+    type=click.Path(),
+    help="configuration to use for deepspeed training",
+    hidden=True,
+)
+@click.option(
+    "--cpu-offload-optim", type=bool, help="if true enables optimizer offload"
+)
+@click.option(
+    "--cpu-offload-params", type=bool, help="if true, enables parameter offload"
+)
+@click.option("--ds-quantize-dtype", type=click.Choice(["nf4", "fp8"]), default=None)
+# below flags are invalid if lora == false
+@click.option("--lora-rank", type=int, help="rank of update matricies")
+@click.option(
+    "--lora-alpha", type=float, help="how influential/strong lors tune will be"
+)
+@click.option("--lora-dropout", type=float, help="dropout for LoRA layers")
+@click.option(
+    "--target-modules",
+    type=str,
+)
+@click.option(
+    "--is-padding-free",
+    type=bool,
+)
+@click.option(
+    "--gpus",
+    "nproc_per_node",
+    type=int,
+    help="this is the number of GPUs to use. This is a torch specific arg and must be called nproc-per-node",
 )
+@click.option("--nnodes", type=int)
+@click.option("--node-rank", type=int)
+@click.option("--rdzv-id", type=int)
+@click.option("--rdzv-endpoint", type=str)
+@click.option("--legacy", hidden=True, type=bool, default=False)
```

Comment:
What is `--legacy` for?

Reply:
@nathan-weinberg `--legacy` is so that we can bring in the new training interface without breaking functionality for existing users who want to use either MacOS or Linux train.

Reply:
you are right, should probably not be hidden